### ISSUE TITLE
Setup in monorepo (`gatsby` dependency is required?)

### ISSUE THREAD

**Issue by stephanschubert** [gatsbyjs/gatsby#10428]
I have a monorepo set up with `lerna` (which shouldn't matter) and moved all (dev)dependencies into the root `package.json`. But it is [required that `gatsby` is added](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-cli/src/create-cli.js#L259-L272) to the package's own `package.json`. 

I'm not running `npm i` inside the (sub-)package—I just added it.

Are there any (legacy?) reasons why this check is required or could it be removed/replaced with a more flexible one?

**Comment by pieh** [comment]
If package (site in case) uses dependency, then it should declare that dependency. If you want to avoid manually installing dependencies in each individual package please take a look at [`lerna bootstrap`](https://github.com/lerna/lerna/tree/master/commands/bootstrap) that will handle that. Another useful thing to consider is using [yarn workspaces](https://yarnpkg.com/lang/en/docs/workspaces/).

Gatsby doesn't really require anything, so there is no other way to determine if current working directory is actually root of gatsby project - if we would remove that check you could run `gatsby build/develop` anywhere and it would probably build empty site which is definitely not great DX

**Comment by stephanschubert** [comment]
I don't use `lerna bootstrap` nor `yarn workspaces` for different reasons. (FYI: Moving all deps into root `package.json` allows me to just run `npm i packages/*` to setup everything up.)

But back to topic: The current check does not ensure the current directory is actually a valid (as in buildable) Gatsby project. A lot of things can be wrong or missing in any case..?

**Comment by sidharthachatterjee** [comment]
Like @pieh mentioned, Gatsby doesn't really require anything so the current check for `gatsby` as a dependency or dev dependency makes sense because it is a pretty strong indicator that the current folder _is_ a Gatsby project if it depends on it. 

If I understand correctly, you want to avoid manually installing dependencies for every individual package (for which you're using `npm i packages/*`)

[lerna bootstrap](https://github.com/lerna/lerna/tree/master/commands/bootstrap) does the same thing:

> npm install all external dependencies of each package.
> Symlink together all Lerna packages that are dependencies of each other.
> npm run prepublish in all bootstrapped packages.
> npm run prepare in all bootstrapped packages.

On a separate note, it is a good practice for individual packages to list their own dependencies so that they are self contained and can work in isolation. 

I'm closing this issue for now but please feel free to reopen if there is anything else.

**Comment by pieh** [comment]
You can also skip using `gatsby-cli` and call build/develop commands directly with some small custom wrapper:
here's entry point for build https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/build.js - you can write very small js file to setup `BuildArgs` (that normally are setup by `cli`)

I do think that tradeoff for not checking local `package.json` is not worth supporting this pretty niche use case - that's the only sanity check we can do :/

**Comment by sidharthachatterjee** [comment]
Thank you for bringing it up anyway, @stephanschubert 

If there's more interest in the usecase, we'll relook at this in the future. In the mean time, happy building. 🙂

**Comment by stephanschubert** [comment]
@pieh Thanks for the hint, I will look into that.

**Comment by josefaidt** [comment]
Hi all, after much trial and error getting a similar setup running I noticed without declaring `gatsby` in the root package.json file gatsby sites in a monorepo will not run - failing resolving modules such as `gatsby/graphql` and `gatsby-plugin-react-helmet`. 

I have a whole issue typed up but figured I could tack on to this issue. I did not notice this note in the documentation, and was wondering if it could be added somewhere?
